internal/watcher/watcher_test.go

```diff
@@ -192,24 +204,131 @@ func TestNewDifferential(t *testing.T) {
 		}
 
 		expected := Differential{
-			ignoredExtensions: getIgnoredExtensions([]string{".swp"}),
+			allowedExtensions: []string{".cfg"},
 			previousChecksum:  previousChecksum,
 			previousContents:  previousContents,
 			client:            http.DefaultClient,
 			url:               "dummy",
 			token:             "2137",
+			diffSender: &RemoteDiffSender{
+				url:    "dummy",
+				token:  "2137",
+				client: http.DefaultClient,
+			},
 		}
 
-		files, _, err := GetAllInDirectory(baseDir, []string{})
+		files, _, err := GetAllInDirectory(baseDir, []string{".cfg"})
 		if err != nil {
 			t.Fatalf("GetAllInDirectory: %v", err)
 		}
 
-		actual, err := NewDifferential([]string{".swp"}, files, http.DefaultClient, "dummy", "2137")
+		actual, err := NewDifferential([]string{".cfg"}, files, http.DefaultClient, "dummy", "2137")
 		if err != nil {
 			t.Fatalf("NewDifferential: %v", err)
 		}
 
 		assert.Equal(t, expected, actual)
 	})
 }
+
+func TestDifferentialIgnore(t *testing.T) {
+	baseDir, err := ioutil.TempDir("", "watcher_test_*")
+	if err != nil {
+		t.Fatalf("ioutil.TempDir: %v", err)
+	}
+
+	defer os.RemoveAll(baseDir)
+
+	file, err := ioutil.TempFile(baseDir, "*")
+	if err != nil {
+		t.Fatalf("ioutil.TempFile: %v", err)
+	}
+	defer file.Close()
+
+	d, err := NewDifferential([]string{".cfg"}, []string{}, http.DefaultClient, "dummy", "2137")
+	if err != nil {
+		t.Fatalf("NewDifferential: %v", err)
+	}
+
+	if err = d.WatchFn(filepath.Join(baseDir, file.Name())); err != nil {
+		t.Fatalf("WatchFn: %v", err)
+	}
+
+	assert.Equal(t, 0, len(d.previousChecksum))
+}
+
+type mockDiffSender struct {
+	called    bool
+	calledMtx sync.Mutex
+}
+
+func (d *mockDiffSender) Send(path string, diff string) error {
+	d.calledMtx.Lock()
+	defer d.calledMtx.Unlock()
+
+	d.called = true
+
+	return nil
+}
+
+func TestDifferentialFiltered(t *testing.T) {
+	baseDir, err := ioutil.TempDir("", "watcher_test_*")
+	if err != nil {
+		t.Fatalf("ioutil.TempDir: %v", err)
+	}
+
+	defer os.RemoveAll(baseDir)
+
+	file, err := ioutil.TempFile(baseDir, "*.cfg")
+	if err != nil {
+		t.Fatalf("ioutil.TempFile: %v", err)
+	}
+	defer file.Close()
+
+	d, err := NewDifferential([]string{".cfg"}, []string{}, http.DefaultClient, "dummy", "2137")
+	if err != nil {
+		t.Fatalf("NewDifferential: %v", err)
+	}
+
+	mockedDiffSender := &mockDiffSender{}
+	d.diffSender = mockedDiffSender
+
+	if err = d.WatchFn(file.Name()); err != nil {
+		t.Fatalf("WatchFn: %v", err)
+	}
+
+	assert.True(t, mockedDiffSender.called)
+	assert.Equal(t, 1, len(d.previousChecksum))
+}
+
+func TestDifferentialLargeFile(t *testing.T) {
+	baseDir, err := ioutil.TempDir("", "watcher_test_*")
+	if err != nil {
+		t.Fatalf("ioutil.TempDir: %v", err)
+	}
+
+	defer os.RemoveAll(baseDir)
+
+	testContent, err := ioutil.ReadFile("testdata/test-large-file.cfg")
```

Comment:
nit: the large file could be generated on demand instead of including it in the repository :)